File: react-mvi-example/src/events/movies-received.ts
import { Event, EventType } from 'src/core/event';
import { Movie } from 'src/entities/movie';

export interface MoviesReceivedPayload {
    movies: Movie[];
    page: number;
    totalPages: number;
}

export class MoviesReceived extends Event<MoviesReceivedPayload> {
    constructor(payload: MoviesReceivedPayload) {
        super(EventType.MoviesReceived, payload);
    }
}

File: android-mvi-example/app/src/main/java/com/servicetitan/mviexample/events/MoviesReceived.kt
package com.servicetitan.mviexample.events

import com.servicetitan.mviexample.core.Event
import com.servicetitan.mviexample.entities.Movie

class MoviesReceived(val payload: Payload) : Event() {
    class Payload(
        val movies: Iterable<Movie>
    )
}